torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -323,8 +323,8 @@ std::string dump_nccl_trace() {
   }
   ncclCommDevIdxMapMutex.unlock();
   for (auto& ncclComm : allNCCLComms) {
-    std::string ncclUniqueIDStr = buildNcclUniqueIdStr(ncclComm->getNcclId());
-    ncclDumpMap[ncclUniqueIDStr] = ncclComm->ncclCommDump();
+    auto commHash = ncclComm->getCommHash();
+    ncclDumpMap[commHash] = ncclComm->ncclCommDump();
```

Comment:
commHash is a uint, but the map key is a string, this is implicit type conversion? better make the type explicit. Also the code needs to work even if NCCL_COMM_GET_UNIQUE_HASH is not defined. In which case, comhash would be 0, and every comm dump will have the same commHash and the dump would be inaccurate?

Reply:
In ncclCommDump function we use commHash, thus the definition of NCCL_COMM_DUMP should always have NCCL_COMM_GET_UNIQUE_HASH defined, and commHash=0 should not be expected here. Also due to this, I changed the map key to uint.

Reply:
NCCL_COMM_GET_UNIQUE_HASH is already defined in NCCL-EXP and we have some functions in both NCCL and PyTorch using commHash even if we don't do ncclCommDump. Thus it would be better to keep both macro: 1) when we do ncclCommDump, commHash will always be valid, 2) when we don't have NCCL_COMM_DUMP defined, functions using commHash can still work.